demos/common/mqtt_demo_helpers/mqtt_demo_helpers.c

```diff
@@ -407,6 +480,146 @@ static TransportSocketStatus_t prvConnectToServerWithBackoffRetries( NetworkCont
 
 /*-----------------------------------------------------------*/
 
+static MQTTStatus_t prvWaitForPacket( MQTTContext_t * pxMQTTContext,
+                                      uint16_t usPacketType )
+{
+    uint8_t ucCount = 0U;
+    MQTTStatus_t xMQTTStatus = MQTTSuccess;
+
+    /* Reset the packet type received. */
+    globalPacketTypeReceived = 0U;
+
+    while( ( globalPacketTypeReceived != usPacketType ) &&
```

Comment:
A comment about reason of calling `MQTT_ProcessLoop` multiple times will be helpful for a reader.
```suggestion
    /* Call MQTT_ProcessLoop multiple times over small timeouts instead of single 
      * a large timeout so that we can unblock immediately on receiving the packet. */
    while( ( globalPacketTypeReceived != usPacketType ) &&
```